ui/hooks/snaps/useTransactionInsightSnaps.js

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { getTransactionOriginCaveat } from '@metamask/snaps-controllers';
+import { handleSnapRequest } from '../../store/actions';
+import { getPermissionSubjectsDeepEqual } from '../../selectors';
+
+const INSIGHT_PERMISSION = 'endowment:transaction-insight';
+
+export function useTransactionInsightSnaps({
+  transaction,
+  chainId,
+  origin,
+  insightSnaps,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-mmi,build-beta)
+  insightSnapId = '',
+  ///: END:ONLY_INCLUDE_IN
+}) {
+  const subjects = useSelector(getPermissionSubjectsDeepEqual);
+
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState(undefined);
+  ///: BEGIN:ONLY_INCLUDE_IN(build-flask)
+  const [hasFetchedV2Insight, setHasFetchedV2Insight] = useState(false);
+  ///: END:ONLY_INCLUDE_IN
+  useEffect(() => {
+    let cancelled = false;
+
+    async function fetchInsight() {
```

Comment:
Two issues:
- In stable the insight is also fetched eagerly, I don't think we want that before we test it in Flask
- In stable the snap insight looks to be refetched every 30ish seconds, this does not happen in prod

Reply:
[06f9bb8](https://github.com/MetaMask/metamask-extension/pull/20554/commits/06f9bb8ba675452c57d0377167bd647bb941e1d4) should address it

Reply:
Yep, stable looks solid now!

Reply:
I'm so confused, when reviewing this today I reproduced the refetching every 30ish seconds again 🤔